enhancements/hypershift/node-lifecycle.md

```diff
@@ -0,0 +1,198 @@
+
+### User Stories
+- As a Service Provider I want to have the ability to managed compute capacity dynamically via declarative API, so I can satisfy end user workloads compute demand:
+  - Scaling.
+  - Autoscaling.
+  - Autorepair.
+- As a Service Provider I want to have the ability upgrade the OS and Kube version of the Nodes in place for baremetal environments, so Hosts don't need to be setup every time.
+- As a Service Provider I want to have the ability upgrade the OS and Kube version of the Nodes by recreating the instances, so Hosts are recycled regularly for security hygiene.
+- As a Service Provider I want to have the ability to declaratively specify different pools of Nodes, so I can setup topologies tolerant to domain failures.
+- As a Service Provider I want to have the ability to declaratively specify different pools of Nodes, so I can allocate workloads with concrete architecture or hardware demands.
+- As a Service Provider I want to have the ability to upgrade the control plane independently of the data plane, so they can satisfy different compliance policies.
+- As a cluster Admin I want to have the ability to manipulate Nodes with Kubernetes high level primitives e.g. taints, labels, etc. Without interacting the underlying infra.
+
+### Goals
+- Provide a consumable API for Node management that satisfies HyperShift form factor (multi-tenancy and control plane/data plane decoupling) supporting the following features:
+  - Scaling.
+  - Auto scaling.
+  - Auto repair.
+  - Auto approval.
+  - OS upgrades.
+  - Config updates.
+  - Management/workload cluster network and infra separation.
+  - Multi-tenancy.
+
+### Non-Goals
+
+## Proposal
+### API
+[NodePool](https://github.com/openshift/hypershift/blob/27c0a432bdc8d702f1cdb2a2f3f25e5ae6fbee7d/api/v1alpha1/nodepool_types.go) is the consumer facing API exposed for Node management.
+Its reason to exist is to preserve our ability to satisfy consumer needs and evolve at our own pace while reusing and relying on battle proven technologies for the implementation details.
+
+### Automated compute capacity management
+#### Cluster API
+Having a NodePool API creates the need for machinery to satisfy their intent. We choose to leverage [CAPI](https://github.com/kubernetes-sigs/cluster-api) as the NodePool implementation.
+It's a project that we have been involved since early stages and contains the learnings from running the `machine.openshift.io` in production since OCP 4 release.
+CAPI is now v1beta1 which gives us the API guarantees we require for customer facing APIs.
+It solves the main problems intrinsic to the HyperShift form factor paradigm in this area: 
+- Cloud agnostic Automated machine management.
+- Awareness of management vs workload cluster.
+- Ability to signal Externally managed infrastructure.
+- Pluggable control plane implementation.
+- Pluggable host bootstrapping implementation.
+
+#### Implementation
```

Comment:
It would be helpful to clarify where each of the components described here runs, especially whether it's in the management cluster or hosted cluster (if those are the right terms). Some seem to be missing, and I'm not sure I understand the descriptions of the ones I noticed. For example, I think when you say "in the control plane namespace" that means it runs in the management cluster in the same namespace where the hosted cluster's control plane is?

If there's another enhancement or architecture document that makes those places clear, you could just link to it or you could provide a short explanation of the locations at the top of this section.

Reply:
Clarified and added to the glossary.

>If there's another enhancement or architecture document that makes those places clear, you could just link to it or you could provide a short explanation of the locations at the top of this section.

Not just yet. This is a step towards that. We are just starting to put together knowledge that today is tribal and scattered.

Reply:
Documenting stuff like this retroactively can be tedious and challenging, so thanks for doing it. What you have here looks like a reasonable start.